### DISCUSSION TITLE
内存占用疑问

### DISCUSSION THREAD

**Discussion by ruixiu** [milvus-io/milvus#18086] [Q&A and General discussion]
milvus2.0.2，6核cpu，32G内存，并发1000写入
1.集合truck_face，2048维向量，入库10万
2.集合truck_face，256维向量，入库90万
根据上述条件：
1）内存占用75%
2）停止写入后，内存会慢慢释放
3）使用attu工具对集合truck_face按id>0查询：内存用完32G

问：1）内存慢慢释放的原理是什么？2）如何正确的估算内存需求？
另：10并发写入6万多，etcd内存占用20%，这是为什么？

**Comment by ruixiu** [comment]
milvus重启后，会多出一个不可用的query node
![image](https://user-images.githubusercontent.com/9940354/177483266-d01c0876-7501-4ccb-bb70-aacf8218d61d.png)

**Reply by xiaofan-luan** [reply]
这个应该是querynode在恢复状态，如果持续卡住的是否是集群未能恢复？

**Comment by yhmo** [comment]
以前1.x版本是用C++写的，内存释放速度很快。2.x是用go写的，内存释放是由go的垃圾回收机制来保证，确实有一定的滞后。
2.0.2上有一些内存使用上的问题，原因有很多，内存泄漏，segment的预留空间过大等等。我们都基本在2.1上做了修复。理想的情况是query node所需的数据量是多大，占用的内存量就应该是多大。在2.1里是占用量比数据量稍微大一点。

重启后多出一个不可用的query node，这个问题似乎没见过，你可以建个issue。

PS： 鼓励用英文提问哈
